### PR TITLE
HIVE-23197

### PR DESCRIPTION
**Pull request by szlta**

### CODE REVIEW

File: llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelLrfuCachePolicy.java [hunk 1]
```diff
@@ -147,7 +152,9 @@ public void notifyLock(LlapCacheableBuffer buffer) {
 
   @Override
   public void notifyUnlock(LlapCacheableBuffer buffer) {
-
+    if (proactiveEvictionEnabled) {
```
Comment (pvary) [inline llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelLrfuCachePolicy.java:155]: Why is this?

Comment (szlta) [reply llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelLrfuCachePolicy.java:155]: notifyUnlock(buffer) is invoked whenever a buffer was accessed (and the read from it has been finished)
This here covers a very rare scenario, details in the test here: https://github.com/apache/hive/pull/1675/files#diff-826be47ae17dbe2de96b3a40cf13fcc40604c5f630fd95d2732a646318ef33d4R307-R311

Comment (pvary) [reply llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelLrfuCachePolicy.java:155]: Maybe add a comment, if we have to push any changes?

Comment (szlta) [reply llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelLrfuCachePolicy.java:155]: ok

File: llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelLrfuCachePolicy.java [hunk 2]
```diff
@@ -240,18 +247,38 @@ public void setEvictionListener(EvictionListener listener) {
...
+            lockedHead.prev = current;
+            lockedHead = current;
+          } else {
+            lockedHead = current;
+            lockedTail = current;
+            current.next = null;
+          }
+
+          current = newCurrent;
+          continue;
```
Comment (pvary) [inline llap-server/src/java/org/apache/hadoop/hive/llap/cache/LowLevelLrfuCachePolicy.java:277]: Have I told you that I hate `continue`?
This is basically an `if/else` in disguise :(

**Review by pvary — APPROVED** [review]

**Review by asinkovits — APPROVED** [review]
LGTM +1 (non-binding)

**Pull request merged**